src/SFML/Graphics/Shader.cpp

```diff
@@ -235,19 +235,19 @@ struct Shader::UniformBinder
 
 
 ////////////////////////////////////////////////////////////
-Shader::Shader() : m_shaderProgram(0), m_currentTexture(-1), m_textures(), m_uniforms()
+void Shader::ShaderProgramDeleter::operator()(const unsigned int shaderProgram)
```

Comment:
SFML doesn't really use top-level CV qualifiers on function parameters; there's also no benefit here.

Reply:
We don't add `const` to value parameters _in the declarations_ because doing so has no effect. This is good. However, adding `const` to the parameter _in the definition_ does matter and is something SFML ought to do. I'll remove this for the sake of consistency but we gain nothing by allowing ourselves to mutate these parameters. Hopefully we can change tact and start adding `const` in more places where it can potentially catch bugs and better expresses our intent.

Reply:
> we gain nothing by allowing ourselves to mutate these parameters

I would agree if const were the default and mutable would need explicit annotating, but at the moment we gain consistency between declaration and definition signature by keeping them in sync.

Furthermore, I think `const` parameters is a bit of a theoretical issue that makes code less readable while hardly preventing bugs in real-life code.

It also opens a whole can of worms like
```cpp
void doStuff(const Texture* const texture) { ... }
```